### ISSUE TITLE
[Dungeon] Threat issues

### ISSUE THREAD

**Issue by ninc** [SunwellWoW/Sunwell-TBC-Bugtracker#1353]
Decription: 
Tanks seems to not do enough threat.

How it works: 
Threat seems to not work correctly. We suspect that abilities that should give extra threat does not do so.
Example 1 (Warrior):
I pulled a pack in The underbog and critted a shield slam. Half a second later our healer took aggro from the mob I critted.
Example 2 (Paladin):
I built a protection paladin with 500 spell dmg (a lot more than most prot pala had back in the day). Even after avenger shield and consecration with rightous fury activated I lost threat to our healer and dps.
Example 3 (Druid):
Problem with threat against single target even with max rage and spamming maul, mangle and lacerate.

How it should work: 
Example 1 (Warrior):
If I use shield slam on a target, that should hold threat for about 3 seconds before I have to swap back and do more threat on that target. If I crit it should be about 5 seconds.
Example 2 (Paladin):
Back in retail TBC I tanked heroic dungeons with t6 geared people with 300 spell damage and there was no threat issues unless a warlock critted.
Example 3 (Druid):
A druid should be able to hold threat by spamming the above combo against most targets. Maybe in a raid scenario with raid buffs threat could be an issue on single target.

Source (you should point out proofs of your report, please give us some source): 
Created all tanks, used the best gear available and played 3x Heroics 2018-11-11. We almost wiped on every pack because of the threat issue.

**Comment by badmannerkid** [comment]
agreed, tank threat seems very low. needs to be brought up. i'm uploading full videos of all instances coming soon

**Comment by arcwizard** [comment]
For such a report you need to calculate threat, compare and then write a bug report. For example: an attack deals 1k damage, have one +100% threat modifier, set up a omen (or take another threat meter) to show a threat without a group, attack a mob with this ability and look at the value in omen, If there is a difference between the calculated value and the value in threat meters, put all this here. Then it will be clear what ability and exactly how it doesn't work. A few days ago i went to the heroic ramparts, didn't notice any serious problems with paladin threat, although i didn’t carry out exact calculations and measurements on this score.

**Comment by neheri** [comment]
I agree with @arcwizard, such crucial PvE feature requires reliable and trustful reports. Opinion is just not enough.

I played on HG which was used as a core for this server and in threat matters everything was perfect. Maybe you are just used to WoTLK when threat was 100x easier to maintain than on TBC?

Anyways, as arcwizard said, there is an easy way to check whether it works:

- as & pally tank and a healer find a mob with large healthpool and which doesn't hit very hard.
- use consecration with righteous fury enabled, calculate how much damage in total it dealt, for example: 100 x 8 = 800 dmg, multiply it by threat modifiers afterwards: 800 x 1.5 (with righteous fury): 1200 or 800 x 1.8 (with righteous fury and talent): 1440. Be sure not to use any auras, white attacks, anything. 

- wait until your health is lowered by at least 5k by the attacking mob. 

- Let healer stand in melee range slowly heal you and and watch when he overtakes the aggro. Melee takes aggro with 110%, ranged 130%. 

Healing gives 0.5 threat per heal, therefore if you want to get 1000 threat you must heal 2000 HP. Overhealing doesn't work.

- If the aggro is taken when the healer has ~950 aggro that means the righteous fury doesn't work.
- if the aggro is taken by the healer with ~1400 it means that rigtheous fury works but talent doesn't,
- if the aggro is taken by the healer with ~1600 that means that righteous fury and talent works.

It all depends on talent set up. Please make sure that healer doesn't have Salvation or any talents having threat reducing factors.

Also bear in mind that when paladin missing health is cured, he receives mana from spiritual attunement which gives threat to paladin.

Testing with warrior and druid could also be done the same way. I was planning to do it myself and will probably do it soon.

Nevertheless this is very important and cannot be subject to an opinion "tanks generate too small threat".

**Comment by ninc** [comment]
Thank you for the feedback.

My experience is from tanking on vanilla servers. However I do not have the
facts to back this up yet. I will try to do some more extensive testing
after work today and update the thread.

Thank you for all your hard work. Looking forward to playing on your server!

**Comment by arcwizard** [comment]
I did some test for paly:
JoR: 
823 holy dmg + 90% (Righteous Fury) = 1563,7 threat. In omen 1,7k threat.
![default](https://user-images.githubusercontent.com/43764742/48364436-16bec100-e6b1-11e8-91c3-df86d40235d8.jpg)

1 melee hit + 1 hit SoR: 
47 physical dmg + (278 holy dmg + 90%) = 575 threat. In omen 575 threat.
![wowscrnshot_111218_203111](https://user-images.githubusercontent.com/43764742/48364785-f9d6bd80-e6b1-11e8-8cf2-b213b07d9d1b.jpg)

Conc(DoT):
534 holy dmg + 90% (Righteous Fury) = 1014,6 threat. In omen 1k threat.
![sda](https://user-images.githubusercontent.com/43764742/48365138-b92b7400-e6b2-11e8-8088-b9379ba25030.jpg)

**Comment by neheri** [comment]
Thank you. However this is still Omen which is just LUA reading buffs, talents and damage dealts and recalculating everything. I am not entirely sure if it reflects the threat accurately. 

While this can be true, threat debugging command would be very useful.

Alternatively you could perform a test with some healer on a mob with large health pool, that would give particular answers.

**Comment by arcwizard** [comment]
ok, same with .debug threat
![231](https://user-images.githubusercontent.com/43764742/48441565-8c4f8d80-e794-11e8-9100-6c478499c552.jpg)
![3223](https://user-images.githubusercontent.com/43764742/48441566-8c4f8d80-e794-11e8-9179-f04d3a8d6b3d.jpg)
![4124](https://user-images.githubusercontent.com/43764742/48441567-8c4f8d80-e794-11e8-974c-c991a99c7f92.jpg)

**Comment by ninc** [comment]
Hi again,

We did some testing tonight and I made a video.
https://youtu.be/xT5apRbMP3M

There are 2 interesting points in the video:
1. At 19 seconds in, the priest Edz pulls aggro from me despite being 23% lower than me on threat. Important to note also is that he moves into melee range for some reason. Maybe that has something to do with it?

2. At 101 seconds in, the warrior Deltisa pulls aggro from me despite only being 1% over me in threat.

**Comment by neheri** [comment]
Hello,

Welcome to TBC, when tanking was actually challenging. 

I remember tanking this boss very well, it was difficult unless you were using spell reflect on his bolts. Have you tested that? I bet that with spell reflect they wouldn't overaggro you.

I don't know your build & gear, however I would consider ~800 TPS to be normal in 5 man. Especially on this boss, you usually run out of range so you cannot spam heroic strike.

Your teammates seem to have Archimonde or Zul'Jin sword, there's no Blessing of Salvation on them so yes, for me it looks alright.

Please notice that threat shown by omen is calculated from combat log, multipled by buff, talents and other modifiers. It can't be 100% accurate. However if you say that he had 1% more threat than you it's very likely that omen just wasn't taking something into consideration (shouts, rage gains?).

Regarding priest - has he used vampiric embrace? I remember back into days that it just generated tons of threat and I think that this can also not be considered by omen.

Obviously it would the best if there was threat debugging command, so you could just check your actual threat on server side after using an ability. 

Being overaggroed by priest having 23% less threat can be caused EITHER by inaccurate omen OR not working defiance talent. This requires further testing.

**Comment by arcwizard** [comment]
In addition to the above written, it is possible that some kind of bug on the side of NPС, or maybe not a bug, for example, if you attack a target with a tank that absorbs your damage, then you will not create a threat.

**Comment by kiterama** [comment]
Today we tested 5man HC Steamvault and Underbog.

What we noticed there is a big issue on threat system. The mobs are not moving to the target which took aggro after the mobs are pulled by our warry mt. It doen'st matter if he bodypulled or used any ability. We could overaggro and the mobs still followed the tank. 
In case the tank moved towards the person which overaggroed and moved into him, the mobs started to hit the player but when the tank moved out, the mob stopped fighting the player with the most aggro.

If needed we can provide few videos with this issue
